### PR TITLE
integration: implement integration tests for docker

### LINKED ISSUE #70: docker: missing integration tests
implement integration tests for docker

### PR DESCRIPTION
**Pull request by devimc**
add integration test for docker

fixes #70

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

File: docker.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	state := strings.Split(stdout, " \n\t")
+	return state[0]
+}
+
+// ContainerExist returns true if any of next cases is true:
+// - ps command shows the container
+// - the VM is running (qemu)
+// else false is returned
+func ContainerExist(id string) bool {
```
Comment (jodh-intel) [inline docker.go]: It would be slightly more readable if this function was called `ContainerExists()`.

File: docker.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	state := GetContainerStatus(id)
+	if state != "" {
+		return true
+	}
+
+	//FIXME: implement function to check if VM is running
```
Comment (jodh-intel) [inline docker.go]: Is there an issue for this?

File: integration/docker/run_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+			LogIfFail(command.Stderr.String())
+
+			Ω(expectedExitCode).Should(Equal(exitCode))
+		},
+		withWorkload("true", 0),
+		withWorkload("false", 1),
+		withWorkload("exit 0", 0),
+		withWorkload("exit 1", 1),
+		withWorkload("exit 15", 15),
+		withWorkload("exit 123", 123),
```
Comment (jodh-intel) [inline integration/docker/run_test.go]: Wow, that's neat! ;)

File: docker.go (hunk 2 above)
Comment (devimc) [inline docker.go]: yes, https://github.com/clearcontainers/tests/issues/69

File: docker.go (hunk 1 above)
Comment (devimc) [inline docker.go]: good catch, thanks

**Comment by devimc** [comment]
depends on https://github.com/clearcontainers/tests/pull/72

File: Makefile [hunk 4]
```diff
@@ -33,9 +33,14 @@ ginkgo:
...
+integration: ginkgo
+	./ginkgo ./integration/docker/ -- -timeout ${TIMEOUT}
+
 # Add crio tests here when https://github.com/clearcontainers/runtime/issues/215
-# is fixed.
-check:	functional
+check:	functional integration
+
+
+check:	functional integration
```
Comment (chavafg) [inline Makefile]: duplicated line with L40

**Comment by GabyCT** [comment]
@devimc could you please change Ω to expect ?

**Comment by devimc** [comment]
@GabyCT sure

**Comment by GabyCT** [comment]
@devimc thanks

**Comment by GabyCT** [comment]
lgtm

**Pull request merged**